slumber/processing/psd_feature.py

```diff
@@ -0,0 +1,124 @@
+from dataclasses import dataclass
+import numpy as np
+from loguru import logger
+from scipy.signal import butter, filtfilt, welch
+from scipy.signal.windows import hamming
+from sklearn.linear_model import HuberRegressor
+
+from slumber import settings
+from slumber.utils.data import Data
+from slumber.processing.transforms import Transform
+
+# ——— modural helper functions ——— #
+def _band_power(pxx: np.ndarray, freq: np.ndarray, band: tuple[float, float]) -> float:
+    lo, hi = band
+    mask = (freq >= lo) & (freq <= hi)
+    return float(pxx[mask].mean()) if mask.any() else 0.0
+
+
+def _spectral_entropy(pxx: np.ndarray) -> float:
+    if pxx.size == 0 or pxx.sum() == 0:
+        return 0.0
+    p_norm = pxx / pxx.sum()
+    se = -np.sum(p_norm * np.log2(p_norm + np.finfo(float).eps))
+    return float(se / np.log2(p_norm.size))
+
+
+def _one_over_f_slope(freq: np.ndarray, pxx: np.ndarray, rng: tuple[float, float] = (2, 20)) -> float:
+    mask = (freq >= rng[0]) & (freq <= rng[1]) & (pxx > 0)
+    if np.count_nonzero(mask) < 2:
+        return 0.0
+    log_f = np.log10(freq[mask]).reshape(-1, 1)
+    log_p = np.log10(pxx[mask])
+    return float(HuberRegressor().fit(log_f, log_p).coef_[0])
+
+# pull defaults from settings
+DEFAULTS = settings['psd_feature']
+
+@dataclass(slots=True)
+class FeatureConfig:
+    fs: int = DEFAULTS['fs']
+    window_s: int = DEFAULTS['window_s']
+    step_s: int = DEFAULTS['step_s']
+    freq_bands: np.ndarray = np.array(DEFAULTS['freq_bands'])
+    bp_lo_hi: tuple[float, float] = tuple(DEFAULTS['bp_lo_hi'])
+    bp_order: int = DEFAULTS['bp_order']
+
+class PSDFeatures(Transform):
+    """
+    Modular Slumber Transform for PSD-based feature extraction.
+    Computes band_power, band_ratio, spectral_entropy, and 1/f slope.
+    """
+    def __init__(self, cfg: FeatureConfig | None = None):
+        self.cfg = cfg or FeatureConfig()
+
+    def __call__(self, data: Data) -> Data:
+        logger.debug(f"PSDFeatures config: {self.cfg}")
+        sig = data.array.squeeze()
```

Comment:
So we are assuming the data object has one channel. So we should check for that.

```suggestion
        if data.n_channels != 1:
             raise ValueError(f"Data must have only 1 channel but got {data.n_channels}.")
             
        array = data.array.squeeze()
```